packages/astro/src/core/config.ts

```diff
@@ -3,8 +3,12 @@ import type { AstroConfig, AstroUserConfig } from '../@types/astro-core';
 import { existsSync } from 'fs';
 import * as colors from 'kleur/colors';
 import path from 'path';
-import { pathToFileURL } from 'url';
+import { pathToFileURL, fileURLToPath } from 'url';
 import { z } from 'zod';
+import load from '@proload/core';
+import pluginTypeScript from '@proload/plugin-typescript';
+
+load.use([pluginTypeScript]);
```

Comment:
As far as TypeScript support goes, I could take it or leave it. It's nice to have but not a blocker for this PR, IMO. Happy to drop this if anyone feels strongly about it.

Reply:
Is there any tradeoff here?

Reply:
Yeah, this pulls in [`@swc-node/register`](https://www.npmjs.com/package/@swc-node/register) as a loader. So that's another native dependency with more noise during install.

I could also quickly write a plugin to use [`tsm`](https://github.com/lukeed/tsm) instead, which would keep us in `esbuild` world.

Reply:
@drwpow OK it took about 10 seconds to write a `tsm` version that relies on `esbuild`. I'm going to swap this with that.

Reply:
Nice! Glad we’re able to stay inside `esbuild` (not that swc mattered too much in this case)